src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java

```diff
@@ -467,4 +472,28 @@ private void throwExceptionIfUserDoesNotExist(DocumentSnapshot userDoc) throws D
             throw new DatabaseAccessException(INVALID_USER, USER_DOES_NOT_EXIST_MESSAGE);
         }
     }
+
+    /**
+     * Updates the FCM token in all the groups the user is subscribed to.
+     * @param token The new FCM token
+     * @param currentToken The current FCM token
+     * @param batch The batch of writes
+     * @throws DatabaseAccessException When the operation is interrupted
+     * @throws DocumentException When the update fails
+     */
+    private void updateTokenInSubscribedGroups(String token, String currentToken, WriteBatch batch)
+        throws DatabaseAccessException, DocumentException {
+        ApiFuture<QuerySnapshot> subscribedGroups = collectionAdapter
+            .getDocumentsWhereArrayContains(Path.ofCollection(Collections.groups), "notification-tokens", currentToken);
+        try {
+            for (QueryDocumentSnapshot group : subscribedGroups.get().getDocuments()) {
+                batch.update(group.getReference(), "FCM", FieldValue.arrayRemove(currentToken), "FCM",
+                    FieldValue.arrayUnion(token));
+            }
+        } catch (InterruptedException e) {
+            throw new DatabaseAccessException("write-failed", e.getMessage());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "write-failed" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436)